grabbit/core.py

```diff
@@ -155,43 +153,81 @@ def __init__(self, path, config=None, dynamic_getters=False,
         self.mandatory = set()
         self.dynamic_getters = dynamic_getters
         self.regex_search = regex_search
-        self.exclude_dir = exclude_dir
+        self.index_regex = None
 
         if config is not None:
-            self._load_config(config)
+            self._load_config(config, **kwargs)
 
-    def _load_config(self, config):
+    def _load_config(self, config, **kwargs):
         if isinstance(config, six.string_types):
             config = json.load(open(config, 'r'))
 
+        for k, v in kwargs.items():
+            if k in config:
+                config[k] = v
+
         for e in config['entities']:
             self.add_entity(**e)
 
+        if 'index' in config:
+            self.inclusion_regex = config['index']
+            if self.inclusion_regex.get('include') and \
+                self.inclusion_regex.get('exclude'):
+                    raise ValueError("You can only define either include or "
+                                     "exclude regex, not both.")
         self.index()
 
+    def _check_inclusions(self, f):
+        ''' Check if file or directory against regexes in config to determine if
+            it should be included in the index '''
+        filename = f if isinstance(f, str) else f.filename
+
+        # If file matches any include regex, then true
+        include_regex = self.inclusion_regex.get('include', [])
+        if include_regex:
+            for regex in include_regex:
+                if re.match(regex, filename):
+                    break
+            else:
+                return False
+
+        # If file matches any excldue regex, then false
+        for regex in self.inclusion_regex.get('exclude', []):
```

Comment:
Well since exclude and include are mutually exclusive, get should just return empty list so it won't actually check anything, but sure stylistically that is arguably better.